lib/date_time/date_time.c

```diff
@@ -320,6 +320,11 @@ int date_time_set(const struct tm *new_date_time)
 {
 	int err = 0;
 
+	if (new_date_time == NULL) {
+		LOG_WRN("The passed in pointer cannot be NULL");
```

Comment:
nit: Personally I'd just use `LOG_ERR` here, all in all it's an error condition.

Reply:
Agree. Will fix.